student/src/main/java/tds/student/configuration/EhCacheConfiguration.java

```diff
@@ -0,0 +1,32 @@
+package tds.student.configuration;
+
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.ehcache.EhCacheCacheManager;
+import org.springframework.cache.ehcache.EhCacheManagerFactoryBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.io.ClassPathResource;
+
+/**
+ * This configuration enables caching with an EhCache in-memory implementation
+ * if redis-caching is not available/configured.
+ */
+@Profile("!redisCaching")
+@Configuration
+public class EhCacheConfiguration {
```

Comment:
I pulled our EhCache configuration out of XML and into this class.  I'd rather switch this based upon a property rather than a profile or lack of a profile, but Spring 3's optional inclusions are a little limited.